### ISSUE TITLE
Parallel (same block) collision issue

### ISSUE THREAD

**Issue by MarkMenzies** [EntelectChallenge/2020-Overdrive#9]
It seems like the 'parallel' collision is not working as per the instructions. In the console log below, both bots would end up on block (y:2, x:12) which (as I understand from the instructions) should mean that Player1 should be (y:1, x:11) and Player2 (y:3, x:11), original lanes (y) and x-1. Instead it seems Player2 has been penalised and has not moved *at all*, still stuck on the original (y:3, x:7) block, whereas Player1 got his full move. Interestingly, both Players *did* receive the speed penalty: 6 down to 3, which could either be due to the actual collision or the fact that (y:2, x:12) is a MUD block.

The console log below as been altered slightly for readability, but all the important parts are still there:

```=======================================
Starting round: 2 
=======================================

player: id:1 position: y:1 x:7 speed:6 state:ACCELERATING ...
opponent: id:2 position: y:3 x:7 speed:6

[░░░░░1░░░░░░▓░░▓░░░▓░░░░░░]
[░░░░░░░░░░▓░░░░░░░░░▓░░▓░░]
[░░░░░2▓░░░░░░░▓░░▓░░░░░░░░]
[░░░░░░░░▓▓░░░░░░░»░░░░░░░░]
=======================================
C;2;TURN_RIGHT
C;2;TURN_LEFT
=======================================
Completed round: 2 
=======================================

=======================================
Starting round: 3 
=======================================

=======================================
round:3
player: id:1 position: y:2 x:12 speed:3 state:HIT_MUD ...
opponent: id:2 position: y:3 x:7 speed:3

[░░░░░░░▓░░▓░░░▓░░░░░░░░░░░]
[░░░░░1░░░░░░░░░▓░░▓░░░░░░░]
[2▓░░░░░░░▓░░▓░░░░░░░░░░»░░]
[░░░▓▓░░░░░░░»░░░░░░░░░▓░░░]
=======================================
C;3;TURN_RIGHT
C;3;TURN_LEFT
=======================================
Completed round: 3 
=======================================

**Comment by styphoiz** [comment]
Was this bug resolved?
I see one of my matches been stuck behind the other player while I am in boost mode, ie. unable to overtake. My bot was stuck behind the other player for a few rounds.
What is the actual resolution here? 
The game rules have not been updated to mention this.

round:17
player: id:1 position: y:2 x:156 speed:15 state:TURNING_RIGHT boosting:true boost-counter:3 powerups: OIL:2
opponent: id:2 position: y:2 x:159 speed:9

[░░░▓░░░░░░░░░░▓░░░░▓░░▓░░░]
[░░░░░1░░2░░░░░░░░░░░░░░░░░]
[░░░░░░░▓░░░░░░░░░░░░░░░░░░]
[░▓░░░▓▓░░░░░░░░░▓░░░░░░░░░]
=======================================
Completed round: 17
=======================================
Starting round: 18
======================================================================================================
round:18
player: id:1 position: y:2 x:167 speed:15 state:ACCELERATING boosting:true boost-counter:2 powerups: OIL:2
opponent: id:2 position: y:2 x:168 speed:9

[░░░▓░░░░▓░░▓░░░░░░░▓░░░▓░░]
[░░░░░12░░░░░░░░░░░░░░░░░▓░]
[░░░░░░░░░░░░░░░▓░░▓░░░░░░░]
[░░░░░▓░░░░░░░░░░░░░▓░░▓░░░]
=======================================
Completed round: 18
=======================================
Starting round: 19
======================================================================================================
round:19
player: id:1 position: y:2 x:176 speed:15 state:ACCELERATING boosting:true boost-counter:1 powerups: OIL:2
opponent: id:2 position: y:2 x:177 speed:9

[░░▓░░░░░░░▓░░░▓░░░░░░░░░▓░]
[░░░░░12░░░░░░░░▓░░░░░»░░░░]
[░░░░░░▓░░▓░░░░░░░░░░░░░░▓░]
[░░░░░░░░░░▓░░▓░░░░░░░░░░░░]
=======================================
Completed round: 19
=======================================
Starting round: 20
======================================================================================================
round:20
player: id:1 position: y:2 x:185 speed:9 state:ACCELERATING boosting:false boost-counter:0 powerups: OIL:2
opponent: id:2 position: y:2 x:186 speed:8

[░▓░░░▓░░░░░░░░░▓░░░░░░░░░░]
[░░░░░12░░░░░»░░░░░░░░░░▓░░]
[▓░░░░░░░░░░░░░░▓░░░▓░░░▓░░]
[░▓░░▓░░░░░░░░░░░░░░▓░░░░░░]
=======================================
Completed round: 20
=======================================
Starting round: 21
======================================================================================================
round:21
player: id:1 position: y:2 x:194 speed:8 state:HIT_MUD boosting:false boost-counter:0 powerups: OIL:2, BOOST:1
opponent: id:2 position: y:2 x:195 speed:9

[░░░░░░▓░░░░░░░░░░░░░░»░░▓░]
[░░░»░12░░░░░░░▓░░░▓░░░░░░░]
[░░░░░░▓░░░▓░░░▓░░░░░░░░░▓░]
[░░░░░░░░░░▓░░░░░░░░░░▓▓░░░]
=======================================
Completed round: 21
The rules state the logic as below for collisions.

Collisions
Commands are processed in parallel, and not sequentially. This means that no bot has a command preference or precedence over the other. Due to this, should after processing bot commands they end up in the same block then, both bots will remain in the lane they were previously, with a speed penalty of -1 being applied.

For Example:

Bot 1 is in position(3,28) with a speed of 6 and issues a ACCELERATE command => future position is (3,36)
Bot 2 is in position(2,31) with a speed of 6 and issues a TURN_RIGHT command => future position is (3,36)
The collision will be resolved by moving bot 1 to (3,35) and bot 2 to (2,35)

**Comment by styphoiz** [comment]
Opening a new issue for the above.
